pkg/fnenv/mock/mock.go

```diff
@@ -145,10 +145,10 @@ type Resolver struct {
 	FnNameIDs map[string]string
 }
 
-func (mf *Resolver) Resolve(fnName string) (string, error) {
-	fnID, ok := mf.FnNameIDs[fnName]
+func (mf *Resolver) Resolve(ref *types.FnRef) (string, error) {
+	fnID, ok := mf.FnNameIDs[ref.ID]
```

Comment:
When using `FnRef` as reference, the problem is that we have to be careful with NPEs, such as here. If you turn FnRef into a value, we don't run into this problem.

FYI, cool thing about protobuf generated structs is that they generate getters (`GetID()`), which have a nil check in them. So even when using pointers, it is safer to use the getters. When using values, this obviously does not matter and directly referencing the field is faster.